dateutil/parser.py

```diff
@@ -499,7 +499,7 @@ def _parse(self, timestr, dayfirst=None, yearfirst=None, fuzzy=False,
                     i += 1
 
                     if (len(ymd) == 3 and len_li in (2, 4)
-                        and (i >= len_l or (l[i] != ':' and
+                        and res.hour is None and (i >= len_l or (l[i] != ':' and
                                             info.hms(l[i]) is None))):
```

Comment:
The indentation here is inconsistent (and won't pass pep8 or pyflakes), please align the parenthesis